pulp_rpm/app/access_policy.py

```diff
@@ -16,25 +16,39 @@ class CopyAccessPolicy(AccessPolicyFromDB):
     def has_perms_to_copy(self, request, view, action):
         """
         Check if the source and destination repository matches the usernames permissions.
+
+        `Fail` the check at first missing permission.
         """
         serializer = view.serializer_class(data=request.data, context={"request": request})
         serializer.is_valid(raise_exception=True)
 
+        all_user_permissions = request.user.get_all_permissions()
+
         for copy_action in serializer.data["config"]:
+            # Get destination repo object by its UUID
             dest_repo_id = copy_action["dest_repo"].split("/")[-2]
             dest_repo = RpmRepository.objects.get(pk=dest_repo_id)
 
+            # Check if user has permissions to destination repository
+            if not (
```

Comment:
Looks good. I think we usually write in this way, even though get_all_permissions will work too, it's just less performant when you look at the queries between has_perm vs get_all_permissions.
```
if not (
                request.user.has_perm("rpm.modify_content_rpmrepository", dest_repo)
                or request.user.has_perm("rpm.modify_content_rpmrepository")
            ):
                return False
```